### PR TITLE
qpdf 11.0.0

### PR DESCRIPTION
**Pull request by ismail**
* Build system switched to cmake
* Explicitly select OpenSSL backend for crypto operations
* Remove livecheck until it can be fixed for the new URL format

### CODE REVIEW

File: Formula/qpdf.rb [hunk 1]
```diff
@@ -20,15 +19,17 @@ class Qpdf < Formula
...
   uses_from_macos "zlib"
 
   def install
-    system "./configure", "--prefix=#{prefix}"
-    system "make"
-    system "make", "install"
+    mkdir "build" do
+      system "cmake", "-DUSE_IMPLICIT_CRYPTO=0", "-DREQUIRE_CRYPTO_OPENSSL=1", "..", *std_cmake_args
+      system "make", "install"
+    end
```
Comment (carlocab) [inline Formula/qpdf.rb:32]: Let's use the template provided by `brew create --cmake`.
```suggestion
    system "cmake", "-S", ".", "-B", "build",
                    "-DUSE_IMPLICIT_CRYPTO=0",
                    "-DREQUIRE_CRYPTO_OPENSSL=1",
                    *std_cmake_args
    system "cmake", "--build, "build"
    system "cmake", "--install", "build"
```

File: Formula/qpdf.rb [hunk 2]
```diff
@@ -20,15 +19,19 @@ class Qpdf < Formula
...
 
   def install
-    system "./configure", "--prefix=#{prefix}"
-    system "make"
-    system "make", "install"
+    system "cmake", "-S", ".", "-B", "build",
+                    "-DUSE_IMPLICIT_CRYPTO=0",
+                    "-DREQUIRE_CRYPTO_OPENSSL=1",
+                    *std_cmake_args
+    system "cmake", "--build, "build"
```
Comment (carlocab) [inline Formula/qpdf.rb:33]:
```suggestion
    system "cmake", "--build", "build"
```
Typo, sorry

Comment (ismail) [reply Formula/qpdf.rb]: Thanks for the speedy review! Merged.

File: Formula/qpdf.rb [hunk 3]
```diff
@@ -1,10 +1,9 @@
...
   desc "Tools for and transforming and inspecting PDF files"
   homepage "https://github.com/qpdf/qpdf"
-  url "https://github.com/qpdf/qpdf/releases/download/release-qpdf-10.6.3/qpdf-10.6.3.tar.gz"
-  sha256 "e8fc23b2a584ea68c963a897515d3eb3129186741dd19d13c86d31fa33493811"
+  url "https://github.com/qpdf/qpdf/releases/download/v11.0.0/qpdf-11.0.0.tar.gz"
+  sha256 "8d99e98893f68f52ca3b579770e7e6f4c96612084d6a0e7e05854a6d631b8fe6"
   license "Apache-2.0"
-  revision 1
 
   livecheck do
```
Comment (carlocab) [inline Formula/qpdf.rb:8]: The `livecheck` block needs to be removed to pick up tags from the new format.

**Comment by carlocab** [comment]
`ocrmypdf` fails with errors like
```
    In file included from src/qpdf/embeddedfiles.cpp:16:
    src/qpdf/pikepdf.h:79:14: error: incomplete type 'QPDFObject' named in nested name specifier
            case QPDFObject::object_type_e::ot_null:
                 ^~~~~~~~~~~~
    /opt/homebrew/include/qpdf/QPDFObjectHandle.hh:57:7: note: forward declaration of 'QPDFObject'
    class QPDFObject;
          ^
```
https://github.com/Homebrew/homebrew-core/runs/8292652870?check_suite_focus=true#step:6:3097

**Comment by ismail** [comment]
`pikepdf` needs a patch, such as:

```
sed -i s,"#include <qpdf/QPDFPageObjectHelper.hh>","#include <qpdf/QPDFPageObjectHelper.hh>\n#include <qpdf/Constants.h>",
sed -i s,QPDFObject::object_type_e::ot_,::ot_,g src/qpdf/{*.h,cpp}
```

but I am not sure how to patch a resource.

**Comment by carlocab** [comment]
Is there an upstream commit we can use? Would we still need the patch if the `pikepdf` resource were updated?

**Comment by carlocab** [comment]
Let's see if updating the resources helps.

**Comment by jbarlow83** [comment]
Upstream for pikepdf/ocrmypdf here. The trouble is we have a circular dependency: pikepdf on macOS links to the Homebrew's qpdf binaries.

It may work to upgrade ocrmypdf's formula to use pikepdf 6.0.0.post1, i.e:
https://files.pythonhosted.org/packages/c0/6b/3ff7fac02840bf97475c11434ddee915a45f2192d028da680c3a4c000fea/pikepdf-6.0.0.post1.tar.gz. Likewise any other users of pikepdf should be moved to 6.0.0.post1.
Then, when qpdf 11 builds, pikepdf 6 should be able to link to it.

The 'sed' patch should be removed - that doesn't address the issue.

Tracking issue for qpdf11/pikepdf6 upgrade issues:
https://github.com/pikepdf/pikepdf/issues/390

**Comment by ismail** [comment]
@jbarlow83 Thank you! That fixed the build issue, and I'll ask for a re-review when !arm builds finish.

P.S: Great to see upstream developers here, thanks for keeping an eye!

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**